### PR TITLE
Fixes NIP-160

### PR DESCRIPTION
**Pull request by larubbio**
For FLW services if we have no state for the user, look for deployed service in all the states in the
circle

(I'm relying on CI to run the tests for me.  I'm a bad dev committing without running them)

### CODE REVIEW

File: api/src/main/java/org/motechproject/nms/api/web/UserController.java [hunk 1]
```diff
@@ -234,6 +234,29 @@ private UserResponse getKilkariResponseUser(Long callingNumber) {
         return kilkariUserResponse;
     }
 
+    private boolean serviceDeplyedInCircle(Service service, Circle circle) {
```
Comment (frankhuster) [inline api/src/main/java/org/motechproject/nms/api/web/UserController.java]: Deplyed > Deployed

**Comment by frankhuster** [comment]
Probably want to fix the typo, but otherwise - LGTM

**Comment by larubbio** [comment]
I should probably add a test for it as well.

File: api/src/main/java/org/motechproject/nms/api/web/UserController.java [hunk 2]
```diff
@@ -234,6 +234,29 @@ private UserResponse getKilkariResponseUser(Long callingNumber) {
...
+        }
+
+        List<State> stateList = circle.getStates();
+        if (stateList == null || stateList.isEmpty()) { // No state available
+            return true;
+        }
+
+        if (stateList.size() == 1) { // only one state available
+            return serviceDeployedInUserState(service, stateList.get(0));
+        }
```
Comment (koshalt) [inline api/src/main/java/org/motechproject/nms/api/web/UserController.java]: this if check could be collapsed into the for loop below right?

Comment (larubbio) [inline api/src/main/java/org/motechproject/nms/api/web/UserController.java]: Maybe.  If so the same should be done in the kilkari version of the code.
In fact maybe I should refactor them.

File: testing/src/test/java/org/motechproject/nms/testing/it/api/UserControllerBundleIT.java [hunk 3]
```diff
@@ -3962,6 +3969,52 @@ public void verifyFT438() throws IOException, InterruptedException {
                 .getStatusCode());
     }
 
+    // Verify if a circle has multiple states and the service is not deployed in any of them than the call
```
Comment (frankhuster) [inline testing/src/test/java/org/motechproject/nms/testing/it/api/UserControllerBundleIT.java]: then

**Comment by frankhuster** [comment]
lgtm

**Pull request merged**